sensor/temp-nct75.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Phoenix-RTOS
+ *
+ * NCT75 i2c Temperature Sensor driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include <sys/msg.h>
+#include <posix/utils.h>
+
+#include <i2c.h>
+
+
+static const uint32_t INVALID_TEMP = -1000 * 1000u;
+
+static const uint8_t DEV_ADDR = 0x48u;      /* NCT75 */
+static const uint8_t TEMP_REG_ADDR = 0x00u; /* stored temperature value - 16bit */
+
+
+/* returns temperature in miliCelsius */
+static int32_t getTemp(void)
+{
+	uint8_t temp_raw[2];
+	int16_t temp_enc;
+
+	if (i2c_regRead(DEV_ADDR, TEMP_REG_ADDR, temp_raw, sizeof(temp_raw)) < 0)
+		return INVALID_TEMP;
+
+	/* 12-bit left-adjusted, 2s complement, every bit is 0.0625 C */
+	temp_enc = (temp_raw[0] << 8) | temp_raw[1];
+	return (temp_enc >> 5) * 125;
+}
+
+
+static void thread(void *arg)
+{
+	uint32_t port = (uint32_t)arg;
+	msg_t msg;
+	unsigned long int rid;
+
+	for (;;) {
+		if (msgRecv(port, &msg, &rid) < 0) {
+			memset(&msg, 0, sizeof(msg));
+			msgRespond(port, &msg, rid);
+			continue;
+		}
+
+		switch (msg.type) {
+			case mtOpen:
+			case mtClose:
+				msg.o.io.err = EOK;
+				break;
+			case mtWrite:
+				msg.o.io.err = -EINVAL;
+				break;
+			case mtRead:
+				/* don't support partial reads, signal EOF */
+				if (msg.i.io.offs > 0) {
+					msg.o.io.err = 0; /* EOF */
+				}
+				else {
+					msg.o.io.err = snprintf(msg.o.data, msg.o.size, "%d\n", getTemp());
+				}
+				break;
+			default:
+				msg.o.io.err = -EINVAL;
+				break;
+		}
+
+		msgRespond(port, &msg, rid);
+	}
+}
+
+
+static void print_usage(const char *progname)
+{
+	printf("Usage: %s [i2c_bus_no <1,4>] [temp_device_no]\n", progname);
+}
+
+
+int main(int argc, char **argv)
+{
+	unsigned int i2c_bus_no, dev_no;
+	uint32_t port;
+	oid_t dev;
+	char devname[sizeof("tempX")];
```

Comment:
As above, `char devname[] = "tempX";` is less error prone and faster.

Reply:
will go for `snprintf`